test/e2e/e2eslow/tls_test.go

```diff
@@ -34,16 +31,7 @@ func TestTLS(t *testing.T) {
 	memberClientTLSSecret := "etcd-server-tls" + suffix
 	operatorClientTLSSecret := "etcd-client-tls" + suffix
 
-	err := e2eutil.PreparePeerTLSSecret(clusterName, f.Namespace, memberPeerTLSSecret)
-	if err != nil {
-		t.Fatal(err)
-	}
-	certsDir, err := ioutil.TempDir("", "etcd-operator-tls-")
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer os.RemoveAll(certsDir)
-	err = e2eutil.PrepareClientTLSSecret(certsDir, clusterName, f.Namespace, memberClientTLSSecret, operatorClientTLSSecret)
+	err := e2eutil.PrepareTLS(clusterName, f.Namespace, memberPeerTLSSecret, memberClientTLSSecret, operatorClientTLSSecret)
```

Comment:
Can you push all the changes onto https://github.com/coreos/etcd-operator/pull/1696 first? Otherwise there is no context and we don't know what should be refactored.

Reply:
to test etcd backup with TLS:
I just need to create a TLS enabled etcd cluster.
create backCR with ClientTLSSecret.

I mainly need a way to create tls enabled cluster.
the two functions i refactored out enables that.
`PrepareTLS` creates secret.
`NewClusterWithTLS` creates the cluster.

Reply:
I understand what it does.
But can you push all the changes onto #1696 first? We need to know the context and changes explicitly.
Even such small refactoring is fine to go in #1696 directly too.

Reply:
kk

Reply:
merged the util functions into #1696